pkg/util/strategicpatch/patch.go

```diff
@@ -963,13 +972,170 @@ func HasConflicts(left, right interface{}) (bool, error) {
 	}
 }
 
+// MergingMapsHaveConflicts returns true if the left and right JSON interface
+// objects overlap with different values in any key. All keys are required to be
+// strings. Since patches of the same Type have congruent keys, this is valid
+// for multiple patch types.
+func MergingMapsHaveConflicts(left, right map[string]interface{}, dataStruct interface{}) (bool, error) {
+	t, err := getTagStructType(dataStruct)
+	if err != nil {
+		return true, err
+	}
+
+	return mergingMapFieldsHaveConflicts(left, right, t, "", "")
+}
+
+func mergingMapFieldsHaveConflicts(
+	left, right interface{},
+	fieldType reflect.Type,
+	fieldPatchStrategy, fieldPatchMergeKey string,
+) (bool, error) {
+	switch leftType := left.(type) {
+	case map[string]interface{}:
+		switch rightType := right.(type) {
+		case map[string]interface{}:
+			leftMarker, okLeft := leftType[directiveMarker]
+			rightMarker, okRight := rightType[directiveMarker]
+			if okLeft || okRight {
```

Comment:
I think of it like this:

1. if one or the other has a directive marker, then we need to consider that before looking at the individual keys, since a directive operates on the whole map.
2. if one has a directive marker and the other doesn't, then we have a conflict, since one is deleting or replacing the whole map, and the other is doing things to individual keys.
3. if they both have markers, but they are not the same directive, then we have a conflict because they're doing different things to the map.
4. if they both have the same directive, then there's no conflict here, and we're done looking at this map.
5. otherwise, check the individual keys.

This seems obtuse enough to warrant a comment.